core/src/main/scala/newtypes.scala

```diff
@@ -70,13 +70,13 @@ object newtypes:
       inline def atLeast(inline bot: A): A        = atLeast(raw(bot))
       inline def atMost(inline top: A): A         = atMost(raw(top))
 
-  trait OpaqueLong[A](using A =:= Long) extends TotalWrapper[A, Long]
-  trait OpaqueDouble[A](using A =:= Double) extends TotalWrapper[A, Double]:
+  class OpaqueLong[A](using A =:= Long) extends TotalWrapper[A, Long]
+  class OpaqueDouble[A](using A =:= Double) extends TotalWrapper[A, Double]:
     extension (inline a: A) inline def +(inline o: Int): A = apply(raw(a) + o)
-  trait OpaqueFloat[A](using A =:= Float) extends TotalWrapper[A, Float]
+  class OpaqueFloat[A](using A =:= Float) extends TotalWrapper[A, Float]
 
   import scala.concurrent.duration.FiniteDuration
-  trait OpaqueDuration[A](using A =:= FiniteDuration) extends TotalWrapper[A, FiniteDuration]
+  class OpaqueDuration[A](using A =:= FiniteDuration) extends TotalWrapper[A, FiniteDuration]
 
   abstract class YesNo[A](using ev: Boolean =:= A):
```

Comment:
should We eliminate the use of implicit evidence here as well?

Reply:
we can't.  it's needed for type safety even though we don't use it.

Reply:
I meant for `YesNo`, which you just did.

Reply:
Yeah I rewrote `YesNo` to be more similar to the others, just extending from `TotalWrapper`